From f7fa6b1b952847137424a42758681a59b3bd8df2 Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Fri, 7 Jun 2013 14:48:07 -0400
Subject: [PATCH 56/99] Ticket 47376 - DESC should not be empty as per RFC 2252
 (ldapv3)

Bug Description:  RFC 2252 (ldapv3), DESC, defined as a qdstring, should not
                  be an empty string.  In the standard schema there are
                  several attributes/objectclasses that have " DESC '' ",
                  which violates the RFC.

Fix Description:  Set all empty DESC's with values.

https://fedorahosted.org/389/ticket/47376

Reviewed by: richm & nkinder (Thanks!!)
(cherry picked from commit 9f73f01a408cef05f112f4ec406949f3d48afe78)
(cherry picked from commit 4a29fe1a3fae5a81007c7dec126f48baa03948e0)
---
 ldap/schema/50ns-mail.ldif |  10 ++--
 ldap/schema/60qmail.ldif   |  20 +++----
 ldap/schema/60radius.ldif  | 132 ++++++++++++++++++++++-----------------------
 ldap/schema/60samba3.ldif  |   2 +-
 4 files changed, 82 insertions(+), 82 deletions(-)

diff --git a/ldap/schema/50ns-mail.ldif b/ldap/schema/50ns-mail.ldif
index 8d06119..27006e0 100644
--- a/ldap/schema/50ns-mail.ldif
+++ b/ldap/schema/50ns-mail.ldif
@@ -73,8 +73,8 @@ attributeTypes: ( 2.16.840.1.113730.3.1.17 NAME ( 'mailForwardingAddress' ) DESC
 attributeTypes: ( 2.16.840.1.113730.3.1.32 NAME ( 'mgrpMsgMaxSize' ) DESC 'Netscape Messaging Server 4.x defined attribute' SYNTAX 1.3.6.1.4.1.1466.115.121.1.15 SINGLE-VALUE X-ORIGIN 'Netscape Messaging Server 4.x' )
 attributeTypes: ( 2.16.840.1.113730.3.1.29 NAME ( 'mgrpMsgRejectText' ) DESC 'Netscape Messaging Server 4.x defined attribute' SYNTAX 1.3.6.1.4.1.1466.115.121.1.26  X-ORIGIN 'Netscape Messaging Server 4.x' )
 attributeTypes: ( 2.16.840.1.113730.3.1.789 NAME ( 'mgrpNoDuplicateChecks' ) DESC 'Netscape Messaging Server 4.x defined attribute' SYNTAX 1.3.6.1.4.1.1466.115.121.1.15 SINGLE-VALUE X-ORIGIN 'Netscape Messaging Server 4.x' )
-objectclasses: ( 2.16.840.1.113730.3.2.3 NAME 'mailRecipient' DESC '' SUP top AUXILIARY MUST ( objectClass ) MAY ( cn $ mail $ mailAlternateAddress $ mailHost $ mailRoutingAddress $ mailAccessDomain $ mailAutoReplyMode $ mailAutoReplyText $ mailDeliveryOption $ mailForwardingAddress $ mailMessageStore $ mailProgramDeliveryInfo $ mailQuota $ multiLineDescription $ uid $ userPassword ) X-ORIGIN 'Netscape Messaging Server 4.x' )
-objectclasses: ( 2.16.840.113730.3.2.37 NAME 'nsMessagingServerUser' DESC '' SUP top AUXILIARY MUST ( objectClass ) MAY ( cn $ mailAccessDomain $ mailAutoReplyMode $ mailAutoReplyText $ mailDeliveryOption $ mailForwardingAddress $ mailMessageStore $ mailProgramDeliveryInfo $ mailQuota $ nsmsgDisallowAccess $ nsmsgNumMsgQuota $ nswmExtendedUserPrefs $ vacationstartdate $ vacationenddate ) X-ORIGIN 'Netscape Messaging Server 4.x' )
-objectclasses: ( 2.16.840.1.113730.3.2.4 NAME 'mailGroup' DESC '' SUP top AUXILIARY MUST ( objectClass ) MAY ( cn $ mail $ mailAlternateAddress $ mailHost $ mailRoutingAddress $ mgrpAddHeader $ mgrpAllowedBroadcaster $ mgrpAllowedDomain $ mgrpApprovePassword $ mgrpBroadcasterPolicy $ mgrpDeliverTo $ mgrpErrorsTo $ mgrpModerator $ mgrpMsgMaxSize $ mgrpMsgRejectAction $ mgrpMsgRejectText $ mgrpNoDuplicateChecks $ mgrpRemoveHeader $ mgrpRFC822MailMember $ owner ) X-ORIGIN 'Netscape Messaging Server 4.x' )
-objectclasses: ( 2.16.840.1.113730.3.2.5 NAME 'groupOfMailEnhancedUniqueNames' DESC '' SUP top AUXILIARY MUST ( objectClass $ cn ) MAY ( businessCategory $ description $ mailEnhancedUniqueMember $ o $ ou $ owner $ seeAlso ) X-ORIGIN 'Netscape Messaging Server 4.x' )
-objectclasses: ( 2.16.840.1.113730.3.2.24 NAME 'netscapeMailServer' DESC '' SUP top AUXILIARY MUST ( objectClass ) X-ORIGIN 'Netscape Messaging Server 4.x' )
+objectclasses: ( 2.16.840.1.113730.3.2.3 NAME 'mailRecipient' DESC 'Netscape Messaging Server 4.x defined objectclass' SUP top AUXILIARY MUST ( objectClass ) MAY ( cn $ mail $ mailAlternateAddress $ mailHost $ mailRoutingAddress $ mailAccessDomain $ mailAutoReplyMode $ mailAutoReplyText $ mailDeliveryOption $ mailForwardingAddress $ mailMessageStore $ mailProgramDeliveryInfo $ mailQuota $ multiLineDescription $ uid $ userPassword ) X-ORIGIN 'Netscape Messaging Server 4.x' )
+objectclasses: ( 2.16.840.113730.3.2.37 NAME 'nsMessagingServerUser' DESC 'Netscape Messaging Server 4.x defined objectclass' SUP top AUXILIARY MUST ( objectClass ) MAY ( cn $ mailAccessDomain $ mailAutoReplyMode $ mailAutoReplyText $ mailDeliveryOption $ mailForwardingAddress $ mailMessageStore $ mailProgramDeliveryInfo $ mailQuota $ nsmsgDisallowAccess $ nsmsgNumMsgQuota $ nswmExtendedUserPrefs $ vacationstartdate $ vacationenddate ) X-ORIGIN 'Netscape Messaging Server 4.x' )
+objectclasses: ( 2.16.840.1.113730.3.2.4 NAME 'mailGroup' DESC 'Netscape Messaging Server 4.x defined objectclass' SUP top AUXILIARY MUST ( objectClass ) MAY ( cn $ mail $ mailAlternateAddress $ mailHost $ mailRoutingAddress $ mgrpAddHeader $ mgrpAllowedBroadcaster $ mgrpAllowedDomain $ mgrpApprovePassword $ mgrpBroadcasterPolicy $ mgrpDeliverTo $ mgrpErrorsTo $ mgrpModerator $ mgrpMsgMaxSize $ mgrpMsgRejectAction $ mgrpMsgRejectText $ mgrpNoDuplicateChecks $ mgrpRemoveHeader $ mgrpRFC822MailMember $ owner ) X-ORIGIN 'Netscape Messaging Server 4.x' )
+objectclasses: ( 2.16.840.1.113730.3.2.5 NAME 'groupOfMailEnhancedUniqueNames' DESC 'Netscape Messaging Server 4.x defined objectclass' SUP top AUXILIARY MUST ( objectClass $ cn ) MAY ( businessCategory $ description $ mailEnhancedUniqueMember $ o $ ou $ owner $ seeAlso ) X-ORIGIN 'Netscape Messaging Server 4.x' )
+objectclasses: ( 2.16.840.1.113730.3.2.24 NAME 'netscapeMailServer' DESC 'Netscape Messaging Server 4.x defined objectclass' SUP top AUXILIARY MUST ( objectClass ) X-ORIGIN 'Netscape Messaging Server 4.x' )
diff --git a/ldap/schema/60qmail.ldif b/ldap/schema/60qmail.ldif
index b1bc391..98d0810 100644
--- a/ldap/schema/60qmail.ldif
+++ b/ldap/schema/60qmail.ldif
@@ -308,7 +308,7 @@ attributeTypes: (
 attributeTypes: (
   1.3.6.1.4.1.7914.1.4.1.1
   NAME 'qladnmanager'
-  DESC ''
+  DESC 'qladnmanager'
   EQUALITY distinguishedNameMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.12
   )
@@ -318,7 +318,7 @@ attributeTypes: (
 attributeTypes: (
   1.3.6.1.4.1.7914.1.4.1.2
   NAME 'qlaDomainList'
-  DESC ''
+  DESC 'qlaDomainList'
   EQUALITY caseIgnoreIA5Match
   SUBSTR caseIgnoreIA5SubstringsMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{256}
@@ -329,7 +329,7 @@ attributeTypes: (
 attributeTypes: (
   1.3.6.1.4.1.7914.1.4.1.3
   NAME 'qlaUidPrefix'
-  DESC ''
+  DESC 'qlaUidPrefix'
   EQUALITY caseIgnoreIA5Match
   SUBSTR caseIgnoreIA5SubstringsMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{256}
@@ -341,7 +341,7 @@ attributeTypes: (
 attributeTypes: (
   1.3.6.1.4.1.7914.1.4.1.4
   NAME 'qlaQmailUid'
-  DESC ''
+  DESC 'qlaQmailUid'
   EQUALITY integerMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.27
   SINGLE-VALUE
@@ -352,7 +352,7 @@ attributeTypes: (
 attributeTypes: (
   1.3.6.1.4.1.7914.1.4.1.5
   NAME 'qlaQmailGid'
-  DESC ''
+  DESC 'qlaQmailGid'
   EQUALITY integerMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.27
   SINGLE-VALUE
@@ -363,7 +363,7 @@ attributeTypes: (
 attributeTypes: (
   1.3.6.1.4.1.7914.1.4.1.6
   NAME 'qlaMailMStorePrefix'
-  DESC ''
+  DESC 'qlaMailMStorePrefix'
   EQUALITY caseIgnoreIA5Match
   SUBSTR caseIgnoreIA5SubstringsMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{256}
@@ -375,7 +375,7 @@ attributeTypes: (
 attributeTypes: (
   1.3.6.1.4.1.7914.1.4.1.7
   NAME 'qlaMailQuotaSize'
-  DESC ''
+  DESC 'qlaMailQuotaSize'
   EQUALITY integerMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.27
   SINGLE-VALUE
@@ -386,7 +386,7 @@ attributeTypes: (
 attributeTypes: (
   1.3.6.1.4.1.7914.1.4.1.8
   NAME 'qlaMailQuotaCount'
-  DESC ''
+  DESC 'qlaMailQuotaCount'
   EQUALITY integerMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.27
   SINGLE-VALUE
@@ -397,7 +397,7 @@ attributeTypes: (
 attributeTypes: (
   1.3.6.1.4.1.7914.1.4.1.9
   NAME 'qlaMailSizeMax'
-  DESC ''
+  DESC 'qlaMailSizeMax'
   EQUALITY integerMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.27
   SINGLE-VALUE
@@ -408,7 +408,7 @@ attributeTypes: (
 attributeTypes: (
   1.3.6.1.4.1.7914.1.4.1.10
   NAME 'qlaMailHostList'
-  DESC ''
+  DESC 'qlaMailHostList'
   EQUALITY caseIgnoreIA5Match
   SUBSTR caseIgnoreIA5SubstringsMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{256}
diff --git a/ldap/schema/60radius.ldif b/ldap/schema/60radius.ldif
index 93a5ba3..3350a9c 100644
--- a/ldap/schema/60radius.ldif
+++ b/ldap/schema/60radius.ldif
@@ -14,7 +14,7 @@ dn: cn=schema
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.1
         NAME 'radiusArapFeatures'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -22,7 +22,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.2
         NAME 'radiusArapSecurity'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -30,7 +30,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.3
         NAME 'radiusArapZoneAccess'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -38,7 +38,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.44
      NAME 'radiusAuthType'
-     DESC ''
+     DESC 'radiusAttribute'
      EQUALITY caseIgnoreIA5Match
      SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
      SINGLE-VALUE
@@ -46,7 +46,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.4
         NAME 'radiusCallbackId'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -54,7 +54,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.5
         NAME 'radiusCallbackNumber'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -62,7 +62,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.6
         NAME 'radiusCalledStationId'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -70,7 +70,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.7
         NAME 'radiusCallingStationId'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -78,14 +78,14 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.8
         NAME 'radiusClass'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.45
      NAME 'radiusClientIPAddress'
-     DESC ''
+     DESC 'radiusAttribute'
      EQUALITY caseIgnoreIA5Match
      SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
      SINGLE-VALUE
@@ -93,14 +93,14 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.9
         NAME 'radiusFilterId'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.10
         NAME 'radiusFramedAppleTalkLink'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -108,14 +108,14 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.11
         NAME 'radiusFramedAppleTalkNetwork'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.12
         NAME 'radiusFramedAppleTalkZone'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -123,14 +123,14 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.13
         NAME 'radiusFramedCompression'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.14
         NAME 'radiusFramedIPAddress'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -138,7 +138,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.15
         NAME 'radiusFramedIPNetmask'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -146,7 +146,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.16
         NAME 'radiusFramedIPXNetwork'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -154,7 +154,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.17
         NAME 'radiusFramedMTU'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -162,7 +162,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.18
         NAME 'radiusFramedProtocol'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -170,14 +170,14 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.19
         NAME 'radiusFramedRoute'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.20
         NAME 'radiusFramedRouting'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -185,14 +185,14 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.46
         NAME 'radiusGroupName'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.47
         NAME 'radiusHint'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -200,14 +200,14 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.48
         NAME 'radiusHuntgroupName'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.21
         NAME 'radiusIdleTimeout'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -215,14 +215,14 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.22
         NAME 'radiusLoginIPHost'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.23
         NAME 'radiusLoginLATGroup'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -230,7 +230,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.24
         NAME 'radiusLoginLATNode'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -238,7 +238,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.25
         NAME 'radiusLoginLATPort'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -246,7 +246,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.26
         NAME 'radiusLoginLATService'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -254,7 +254,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.27
         NAME 'radiusLoginService'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -262,7 +262,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.28
         NAME 'radiusLoginTCPPort'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -270,7 +270,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.29
         NAME 'radiusPasswordRetry'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -278,7 +278,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.30
         NAME 'radiusPortLimit'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -286,7 +286,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.49
         NAME 'radiusProfileDn'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY distinguishedNameMatch
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.12
         SINGLE-VALUE
@@ -294,7 +294,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.31
         NAME 'radiusPrompt'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -302,7 +302,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.50
         NAME 'radiusProxyToRealm'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -310,7 +310,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.51
         NAME 'radiusReplicateToRealm'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -318,7 +318,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.52
         NAME 'radiusRealm'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -326,7 +326,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.32
         NAME 'radiusServiceType'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -334,7 +334,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.33
         NAME 'radiusSessionTimeout'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -342,7 +342,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.34
         NAME 'radiusTerminationAction'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -350,21 +350,21 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.35
         NAME 'radiusTunnelAssignmentId'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.36
         NAME 'radiusTunnelMediumType'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.37
         NAME 'radiusTunnelPassword'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -372,42 +372,42 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.38
         NAME 'radiusTunnelPreference'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.39
         NAME 'radiusTunnelPrivateGroupId'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.40
         NAME 'radiusTunnelServerEndpoint'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.41
         NAME 'radiusTunnelType'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.42
         NAME 'radiusVSA'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.43
         NAME 'radiusTunnelClientEndpoint'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
@@ -415,14 +415,14 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.53
         NAME 'radiusSimultaneousUse'
-        DESC ''
+        DESC 'radiusAttribute'
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.27
         SINGLE-VALUE
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.54
         NAME 'radiusLoginTime'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -430,7 +430,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.55
         NAME 'radiusUserCategory'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -438,14 +438,14 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.56
         NAME 'radiusStripUserName'
-        DESC ''
+        DESC 'radiusAttribute'
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.7
         SINGLE-VALUE
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.57
         NAME 'dialupAccess'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -453,7 +453,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.58
         NAME 'radiusExpiration'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -461,21 +461,21 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.59
         NAME 'radiusCheckItem'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.60
         NAME 'radiusReplyItem'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.61
         NAME 'radiusNASIpAddress'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -483,7 +483,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.62
         NAME 'radiusReplyMessage'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
@@ -491,7 +491,7 @@ objectClasses:
     ( 1.3.6.1.4.1.3317.4.3.2.1
         NAME 'radiusprofile'
         SUP top AUXILIARY
-        DESC ''
+        DESC 'radiusObjectclass'
         MUST uid
         MAY ( radiusArapFeatures $ radiusArapSecurity $ radiusArapZoneAccess $
             radiusAuthType $ radiusCallbackId $ radiusCallbackNumber $
@@ -529,7 +529,7 @@ objectClasses:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.64
         NAME 'radiusClientSecret'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -537,7 +537,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.65
         NAME 'radiusClientNASType'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
         SINGLE-VALUE
@@ -545,7 +545,7 @@ attributeTypes:
 attributeTypes:
     ( 1.3.6.1.4.1.3317.4.3.1.66
         NAME 'radiusClientShortName'
-        DESC ''
+        DESC 'radiusAttribute'
         EQUALITY caseIgnoreIA5Match
         SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
     )
diff --git a/ldap/schema/60samba3.ldif b/ldap/schema/60samba3.ldif
index 5dd5f31..91c17be 100644
--- a/ldap/schema/60samba3.ldif
+++ b/ldap/schema/60samba3.ldif
@@ -204,7 +204,7 @@ attributeTypes: (
 attributeTypes: (
   1.3.6.1.4.1.7165.2.1.47
   NAME 'sambaMungedDial'
-  DESC ''
+  DESC 'sambaMungedDial'
   EQUALITY caseExactMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.15{1050}
   )
-- 
1.8.1.4

